proposals/lip-0052.md

```diff
@@ -679,6 +671,179 @@ else: # chainID == sendingChainID
+                        "dataType": "uint32",
+                        "fieldNumber": 3
+                    },
+                    "attributes": {
+                        "dataType": "bytes",
+                        "fieldNumber": 4
+                    }
+                }
+            }
+        },
+        "collectionSubstore": {
+            "type": "array",
+            "fieldNumber": 2,
+            "items": {
+                "type": "object",
+                "required": ["collection", "nextAvailableIndex"],
+                "properties": {
+                    "collection": {
+                        "dataType": "uint32",
+                        "fieldNumber": 1
+                    },
+                    "nextAvailableIndex": {
+                        "dataType": "uint64",
+                        "fieldNumber": 2
+                    }
+                }
+            }
+        },
+        "availableCollectionSubstore": {
+            "dataType": "uint32",
+            "fieldNumber": 3
+        },
+        "terminatedEscrowSubstore": {
+            "type": "array",
+            "fieldNumber": 4,
+            "items": {
+                "type": "uint32"
+            }
+        }
+    }
+}
+```
+
+#### Genesis State Initialization
+
+During the genesis state initialization stage, the following steps are executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets for the NFT module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisNFTStoreSchema` schema, given above.
+
+* Initial checks:
+   * Across all elements of the `NFTSubstore` array, all values given for `NTFID` must be unique. 
+   * For all elements of the `NFTSubstore` array, values given for `chainID` must have either length 20 (representing a user address) or 4 (representing a chain ID).
+   * The `NFTSubstore` must be in lexicographical order of `owner`. 
+      * For a given `owner`, the entries must be in ascending order of `tokenID.chainID`. 
+      * For a given `owner` and `tokenID.chainID`, the entries must be in ascending order of `tokenID.collection`. 
+      * For a given `owner`, `tokenID.chainID` and `tokenID.collection` the entries must be in ascending order of `tokenID.index`.
+   * For all elements of the `NFTSubstore` array, if the value of `owner` has length 4 (representing an escrowed NFT) then the corresponding `NFTID.chainID` must equal `CHAIN_ID_ALIAS_NATIVE`.
+   * For all elements of the `NFTSubstore` array, if the value of `attributes` must have length less than or equal to `MAX_BYTE_LENGTH_ATTRIBUTES`.
+   * Accross all element of the `collectionSubstore` array, all values given for `collection` must be unique.
+   * The `collectionSubstore` array must be in ascending order of `collection`.
+   * The `terminatedEscrowSubstore` array must be in ascending order. 
```

Comment:
I guess for `terminatedEscrowSubstore` we should also require uniqueness (although the state would be the same).

Reply:
I don't understand what you mean here.